src/plugins/intel_gpu/src/graph/shape_of.cpp

```diff
@@ -33,14 +33,30 @@ layout shape_of_inst::calc_output_layout(shape_of_node const& node, kernel_impl_
     return layout{dt, format::bfyx, out_size};
 }
 
+template<typename ShapeType>
+std::vector<layout> shape_of_inst::calc_output_layouts(shape_of_node const& /*node*/, const kernel_impl_params& impl_param) {
+    auto prim = impl_param.typed_desc<shape_of>();
+
+    data_types output_dt = prim->output_data_type.value_or(data_types::i32);
+    if (impl_param.has_fused_primitives()) {
+        output_dt = impl_param.get_fused_output_layout().data_type;
+    }
+
+    auto in_shape = impl_param.get_input_layout(0).get<ShapeType>();
```

Comment:
should return dynamic layout if input rank is dynamic?

Reply:
Dynamic rank won't be supported in the near future